src/praisonai-agents/tests/test_parallel_tools.py

```
@@ -0,0 +1,203 @@
+    # Test parallel execution
+    print("\n--- Parallel Execution ---")
+    parallel_start = time.time()
+    par_executor = create_tool_call_executor(parallel=True, max_workers=3)
+    par_results = par_executor.execute_batch(tool_calls, mock_execute_tool)
+    parallel_time = time.time() - parallel_start
+    print(f"Parallel execution took: {parallel_time:.2f}s")
+    print(f"Results: {len(par_results)} tools executed")
+    
+    # Verify results are identical and in correct order
+    assert len(seq_results) == len(par_results), "Result counts should match"
+    for i, (seq_result, par_result) in enumerate(zip(seq_results, par_results)):
```

Ruff (0.15.10)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)